### PR TITLE
sys-cluster/spark-bin: Add version 3.1.2

### PR DESCRIPTION
**Pull request by trozamon**
Package-Manager: Portage-3.0.20, Repoman-3.0.3
Signed-off-by: Alec Ten Harmsel <alec@alectenharmsel.com>

### CODE REVIEW

File: sys-cluster/spark-bin/spark-bin-3.1.2.ebuild [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	einfo
+	einfo "Spark is now slotted. You have installed Spark ${SLOT}."
+	einfo
+	einfo "Make sure to add /usr/lib/spark-${SLOT}/{bin,sbin} directories"
+	einfo "to your PATH in order to run Spark shell scripts:"
+	einfo
+	einfo "$ export PATH=\$PATH:/usr/lib/spark-${SLOT}/bin"
+	einfo "$ export PATH=\$PATH:/usr/lib/spark-${SLOT}/sbin"
+	einfo
+}
```
Comment (juippis) [inline sys-cluster/spark-bin/spark-bin-3.1.2.ebuild:71]: Could we instead install a new */etc/env.d* file for this? 
Check https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-lang/rust/rust-1.54.0.ebuild#n616 for example or 
https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-devel/llvm/llvm-12.0.1.ebuild#n499

File: sys-cluster/spark-bin/spark-bin-3.1.2.ebuild [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	)
+
+	local s
+	for s in "${SPARK_SCRIPTS[@]}"; do
+		dobin "${s}"
+	done
+
+	insinto usr/lib/spark-${SLOT}
+
+	local SPARK_DIRS=( conf jars python sbin yarn )
```
Comment (juippis) [inline sys-cluster/spark-bin/spark-bin-3.1.2.ebuild:51]: We name local variables in smallcase lettering to differentiate from global ones, that are written in uppercase.

File: sys-cluster/spark-bin/spark-bin-3.1.2.ebuild [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+DOCS=( LICENSE NOTICE README.md RELEASE )
+
+# Nothing to compile here.
+src_compile() { :; }
+
+src_install() {
+	dodir usr/lib/spark-${SLOT}
+	into usr/lib/spark-${SLOT}
+
+	local SPARK_SCRIPTS=(
```
Comment (juippis) [inline sys-cluster/spark-bin/spark-bin-3.1.2.ebuild:33]: Same here.

**Comment by trozamon** [comment]
Thank you for the feedback - I will address it soon.

**Review by juippis — APPROVED** [review]
Thanks, seems to work for me! Although I wonder a bit about the necessity of these slots.

**Pull request closed without merging**